src/cli/setup_mode.rs

```diff
@@ -97,7 +100,11 @@ pub fn main() -> Result<utils::ExitCode> {
             write!(process().stdout().lock(), "{e}")?;
             return Ok(utils::ExitCode(0));
         }
-        Err(e) => return Err(e.into()),
+        Err(e) => {
```

Comment:
I believe that's not the right place to add the handling logic for `rustup update`. Have you actually tested the code that you added?

Please have a look at the `rustup update` handling logic [here](https://github.com/rust-lang/rustup/blob/66b7a442c074c5cd597ea1436e1bafb0807b46e0/src/cli/rustup_mode.rs#L930), and find out where that error was thrown, and then add an error message using the `info!()` macro.